liota/lib/transports/mqtt.py

```diff
@@ -320,12 +314,27 @@ def subscribe(self, topic, qos, callback):
         :return:
         """
         try:
+            if [topic,qos,callback] not in self.sub_list:
+                self.sub_list.append([topic,qos,callback])
             subscribe_response = self._paho_client.subscribe(topic, qos)
             self._paho_client.message_callback_add(topic, callback)
             log.info("Topic subscribed with information: " + str(subscribe_response))
         except Exception:
             log.exception("MQTT subscribe exception traceback..")
 
+    def unsubscribe(self, topic):
+        """
+        Unsubscribes to a topic
+
+        :param topic: Unsubscribe topic
+        :return:
+        """
+        try:
+            unsubscribe_response = self._paho_client.unsubscribe(topic)
```

Comment:
I thought the whole purpose of implementing unsubscribe() was to remove it from sub_list?

Reply:
It has already been implemented post Glen`s comment.